lib/rubocop/cop/rails/root_public_path.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Favor `Rails.public_path` over `Rails.root` with `'public'`
+      #
+      # @example
+      #   # bad
+      #   Rails.root.join('public')
+      #   Rails.root.join('public/file.pdf')
+      #   Rails.root.join('public', 'file.pdf')
+      #
+      #   # good
+      #   Rails.public_path
+      #   Rails.public_path.join('file.pdf')
+      #   Rails.public_path.join('file.pdf')
+      #
+      class RootPublicPath < Base
+        extend AutoCorrector
+
+        MSG = 'Use `Rails.public_path`.'
+
+        RESTRICT_ON_SEND = %i[join].to_set.freeze
+
+        PATTERN = %r{^public(/|$)}.freeze
```

Comment:
`\A` and `\z` would be preferable if it is unexpecting a match to a multiline.

```suggestion
        PATTERN = %r{\Apublic(/|\z)}.freeze
```